### PR TITLE
VEN-48 & VEN-49 & VEN-51 | Winter Storage Areas

### PR DESCRIPTION
**Pull request by Nizar-Rahme**
The main changes are:
- Extracting GraphQL queries and some props like steps and others from pages components to their correspondent containers
- Creating containers for Winter Storage pages
- Adding Query and Mutation for Winter Storage Areas
- Generating TS types from GraphQL queries using Apollo CLI tool and fixing our types accordingly

### CODE REVIEW

File: src/components/App.tsx [hunk 1]
```diff
@@ -8,19 +8,29 @@ import FormPage from './pages/containers/FormPageContainer';
...
 
+import SelectedAreasPage from './pages/containers/SelectedAreasPageContainer';
+import WinterBerthPage from './pages/containers/WinterBerthPageContainer';
+import WinterFormPage from './pages/containers/WinterFormPageContainer';
+
 export interface Props {
   locale: 'fi' | 'en' | 'sv';
 }
 
+const localeParam = ':locale(fi|en|sv)';
```
Comment (midori0507) [inline src/components/App.tsx]: But why?

File: src/components/berths/Berth/BerthDetails/index.tsx [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
+
+const BerthDetails = ({ iconName, available, value, titleId }: Props) => {
   return (
     <div
       className={classNames('vene-berth__details', {
         'vene-berth__details-not-available': !available
       })}
     >
-      {value && <div className="vene-berth__details__value">{value}</div>}
+      {value !== undefined && <div className="vene-berth__details__value">{value || '-'}</div>}
```
Comment (midori0507) [inline src/components/berths/Berth/BerthDetails/index.tsx]: Isn't `value` is good enough, this check is not valid

File: src/components/berths/Berth/index.tsx [hunk 3]
```diff
@@ -40,6 +41,110 @@ class Berth extends Component<Props, State> {
...
+          />,
+          <BerthDetails
+            key="summerStorageForTrailers"
+            available={berth.summerStorageForTrailers}
+            iconName="dollyEmpty"
+            titleId="page.berths.storage_for_docking_equip"
+          />
+        ];
+      default:
+        return [];
```
Comment (midori0507) [inline src/components/berths/Berth/index.tsx]: should fallback to one of the option ?

File: src/components/berths/Berth/index.tsx [hunk 4]
```diff
@@ -54,7 +159,8 @@ class Berth extends Component<Props, State> {
                 id="error.message.invalid_berth"
                 isOpen={selected && excluded}
               />
-              <Image src={berth.imageFile} alt={berth.name} />
+              {/* TODO: add palceholder image */}
```
Comment (midori0507) [inline src/components/berths/Berth/index.tsx]: Typo `placeholder`

File: src/components/forms/fragments/overview/BoatTypeAndModel.tsx [hunk 5]
```diff
@@ -11,7 +11,7 @@ type Props = {
   InjectedIntlProps;
 
 const BoatTypeAndModel = ({ boatTypeId, boatModel, boatTypes }: Props) => {
-  const boatType = boatTypes.find(type => type.id === boatTypeId);
+  const boatType = boatTypes && boatTypes.find(type => !!type && type.id === boatTypeId);
```
Comment (midori0507) [inline src/components/forms/fragments/overview/BoatTypeAndModel.tsx]: should using lodash `isEmpty` for boatTypes, remove `!!type`

File: src/components/forms/tabs/RegisteredBoat.tsx [hunk 6]
```diff
@@ -30,11 +29,10 @@ export default ({ prefix, ShowBigShipsForm, boatTypes }: Props) => (
         <FormattedMessage tagName="p" id="form.big_ship.text.inspection_and_insurance" />
       </div>
     )}
-    <FormattedMessage tagName="h3" id="form.registered.header.additional_info" />
-
     <BoatInfo />
-    <FormattedMessage tagName="h3" id="form.registered.header.accessibility" />
 
-    <Accessibility />
+    {mode === 'berth' && <Accessibility />}
```
Comment (midori0507) [inline src/components/forms/tabs/RegisteredBoat.tsx]: `berth` as constant pls, can be local constant, or under `/constants`

File: src/components/forms/tabs/UnRegisteredBoat.tsx [hunk 7]
```diff
@@ -1,23 +1,21 @@
...
 
-export default ({ prefix, boatTypes }: Props) => (
+export default ({ mode, boatTypes }: Props) => (
   <Container>
-    <FormattedMessage tagName="h3" id="form.unregistered.header.title" />
     <UnRegisteredBoatDetails boatTypes={boatTypes} />
     <BoatInfo />
-    <FormattedMessage tagName="h3" id="form.registered.header.accessibility" />
-    <Accessibility />
+    {mode === 'berth' && <Accessibility />}
```
Comment (midori0507) [inline src/components/forms/tabs/UnRegisteredBoat.tsx]: constant

File: src/components/pages/containers/FormPageContainer.tsx [hunk 8]
```diff
@@ -13,9 +24,136 @@ interface Props {
...
+  const [currTab, setTab] = useState('');
+  const [tabs, setTabs] = useState(['registered_boat', 'private_person', 'overview']);
+
+  useEffect(() => {
+    const currStep = Math.max(0, findIndex(mapSteps, s => s.includes(tab)));
+    setStep(currStep);
+    setTab(tab || mapSteps[currStep][0]);
+  });
+
+  const steps = [
```
Comment (midori0507) [inline src/components/pages/containers/FormPageContainer.tsx]: should export this array, also same for winter if possible

File: src/components/pages/containers/SelectedAreasPageContainer.tsx [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+  selectedBerths: Berths;
+  selectedServices: SelectedServices;
+  deselectBerth: Function;
+  moveUp: Function;
+  moveDown: Function;
+  localePush: Function;
+  values: {};
+}
+
+const steps = [
```
Comment (midori0507) [inline src/components/pages/containers/SelectedAreasPageContainer.tsx]: export it and re import here

File: src/components/pages/containers/SelectedBerthPageContainer.tsx [hunk 10]
```diff
@@ -22,6 +27,72 @@ interface Props {
   initialValues: {};
 }
 
+const steps = [
```
Comment (midori0507) [inline src/components/pages/containers/SelectedBerthPageContainer.tsx]: Export and re import

File: src/components/pages/containers/WinterBerthPageContainer.tsx [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+  selectBerth: Function;
+  deselectBerth: Function;
+  selectService: Function;
+  deselectService: Function;
+  onSubmit: Function;
+  localePush: Function;
+}
+
+const BerthPageContainer = (props: Props) => {
+  const steps = [
```
Comment (midori0507) [inline src/components/pages/containers/WinterBerthPageContainer.tsx]: Export

File: src/components/pages/containers/WinterFormPageContainer.tsx [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+  const [currTab, setTab] = useState('');
+  const [tabs, setTabs] = useState(['registered_boat', 'private_person', 'overview']);
+
+  useEffect(() => {
+    const currStep = Math.max(0, findIndex(mapSteps, s => s.includes(tab)));
+    setStep(currStep);
+    setTab(tab || mapSteps[currStep][0]);
+  });
+
+  const steps = [
```
Comment (midori0507) [inline src/components/pages/containers/WinterFormPageContainer.tsx]: Export

File: src/components/App.tsx (hunk 1 above)
Comment (Nizar-Rahme) [reply src/components/App.tsx]: So it doesn't accept other values and goes to "not-found" page

File: src/components/berths/Berth/BerthDetails/index.tsx (hunk 2 above)
Comment (Nizar-Rahme) [reply src/components/berths/Berth/BerthDetails/index.tsx]: If the value is provided we should display it in any case, meaning, even if it comes as `null` from GraphQL we should render `-` instead.
We only omit this value when we don't provide it to the component.

Comment (midori0507) [reply src/components/berths/Berth/BerthDetails/index.tsx]: `value ? [Component] : '-'`

Comment (Nizar-Rahme) [reply src/components/berths/Berth/BerthDetails/index.tsx]: This condition will fail when `value === null`

File: src/components/forms/tabs/RegisteredBoat.tsx (hunk 6 above)
Comment (Nizar-Rahme) [reply src/components/forms/tabs/RegisteredBoat.tsx]: Changed it, I was relying on TS :D

File: src/components/pages/containers/FormPageContainer.tsx (hunk 8 above)
Comment (Nizar-Rahme) [reply src/components/pages/containers/FormPageContainer.tsx]: I will fix it under this ticket [VEN-82](https://helsinkisolutionoffice.atlassian.net/browse/VEN-82)

**Pull request merged**